Core/clim-basic/utils.lisp

```diff
@@ -743,3 +728,195 @@ a flag CLOSED is T then beginning and end of the list are consecutive too."
      ((and *print-pretty* *print-readably*)
       (simple-pprint-object ,sink ,self))
      (t ,@body)))
+
+
+
+(defun %line-breaks-1 (string width initial-offset margin start end)
+  (collect (break-line)
+    (macrolet ((split (step-form)
+                 `(let* ((current-margin (- margin initial-offset))
+                         (split start)
+                         (initial-break ,step-form))
+                    (maxf initial-break split)
+                    (when (and (= initial-break split)
+                               (<= initial-offset 0))
+                      (setf initial-break (1+ split)))
+                    (setf current-margin margin)
+                    (do ((split initial-break
+                                (max ,step-form (1+ split))))
+                        ((>= split end))
+                      (if-let ((pos (position #\space string
+                                              :start (min split end)
+                                              :end end
+                                              :test-not #'char=)))
+                        (break-line (setf split pos))
+                        (return-from %line-breaks-1 (break-line)))))))
+      (etypecase width
+        (number   (split (+ split (floor current-margin width))))
+        (function (split (bisect split (1+ end) ; we split *after* the string
+                                 (lambda (index)
+                                   (<= (funcall width string split index)
+                                       current-margin)))))))
+    (break-line)))
+
+;;; Break greedily on the break opportunity. If a single word doesn't fit in the
+;;; line and that line is shorter than an empty line then we break it right
+;;; away. If the line is longer or equal to an empty one then we break this word
+;;; by a character. If there are multiple #\space characters at the line
+;;; beginning they are kept in the previous line.
+(defun %line-breaks-2 (string width initial-offset margin start end opportunities
+                       &aux (width-fn (etypecase width
+                                        (function width)
+                                        (number (lambda (string start end)
+                                                  (declare (ignore string))
+                                                  (* width (- end start)))))))
+  (labels ((skip-whitespace (string index)
+             (and index (position #\space string :start index :test-not #'char=)))
+           (breaks-rec (offset margin start end next-opportunity-index
```

Comment:
As far as I can tell, all recursive calls keep the initial `margin`, so this parameter is not needed.

Reply:
addressed in the newest commit